apps/api/src/app/shared/framework/types/headers.types.ts

```diff
@@ -0,0 +1,28 @@
+import { ApiHeaderOptions } from '@nestjs/swagger';
+
+export enum RequestHeaderKeysEnum {
+  AUTHORIZATION = 'Authorization',
+  USER_AGENT = 'User-Agent',
+}
+
+export enum ResponseHeaderKeysEnum {
+  CONTENT_TYPE = 'Content-Type',
+  RATE_LIMIT_REMAINING = 'RateLimit-Remaining',
+  RATE_LIMIT_LIMIT = 'RateLimit-Limit',
+  RATE_LIMIT_RESET = 'RateLimit-Reset',
+  RATE_LIMIT_POLICY = 'RateLimit-Policy',
+  RETRY_AFTER = 'Retry-After',
+  IDEMPOTENCY_KEY = 'Idempotency-Key',
+  IDEMPOTENCY_REPLAY = 'Idempotency-Replay',
+  LINK = 'Link',
+}
```

Comment:
The intent here is to have a single location for all headers to live as an enum, use-cases and framework capabilities that require them can import and use them. This will help keep naming consistent.

Reply:
@Cliftonz I will also make a Linear ticket to update per use-case implementation of HTTP headers to consume these shared headers instead, to enable better consistency and provide more examples to repeat.